SAMZA-1991: Samza-sql: Implement DSL converter plugin manager

atoomula opened the pull request:
- Introduced DSLConverterPluginManager to invoke the right DSL converter based on the file extension.
- Moved SamzaSqlQueryParser and SqlFileParser to samzasql dsl package

vjagadish1989 commented on samza-sql/src/main/java/org/apache/samza/sql/dsls/DslConverterPluginManager.java:
```diff
@@ -0,0 +1,79 @@
+    DslConverterFactory dslConverterFactory;
+    try {
+      Class<?> clazz = Class.forName(dslConverterFactoryClass);
+      dslConverterFactory = (DslConverterFactory) clazz.newInstance();
```
For greater flexibility, can DSLConverters take in the URL of the DSL they need to translate? "string" seems too low-level an abstraction to deal with.

vjagadish1989 commented on samza-sql/src/main/java/org/apache/samza/sql/dsls/DslConverterPluginManager.java:
```diff
@@ -0,0 +1,79 @@
+  static public DslConverter create(Config config) {
+    // Default dsl file extension is sql. The default is applied when the sqlFile config is not set.
+    String dslFileExtension = "sql";
+
```
If this is just class-loading utils, does it need a class or can it be inlined at the call-site?

vjagadish1989 commented:
@atoomula is this PR still required?

atoomula commented:
> @atoomula is this PR still required?

Yes @vjagadish1989 . It went down in priority since the time I sent it out for review. But I will work towards addressing the comments and committing next week.
